### DISCUSSION TITLE
REINFORCE in DI-engine?

### DISCUSSION THREAD

**Discussion by cpwan** [opendilab/DI-engine#517] [Ideas]
The REINFORCE algorithm is a classical policy gradient method. It has been implemented in some other RL libraries, such as [RLlib](https://github.com/ray-project/ray/blob/master/rllib/algorithms/pg/pg.py) and [Tianshou](https://github.com/thu-ml/tianshou/blob/master/tianshou/policy/modelfree/pg.py). I wonder if we can also have it in DI-engine.

**Comment by PaParaZz1** [comment]
It is not difficult to implement naive policy gradient in DI-engine, but I want to know why you need it? We didn't add naive pg before because it shows poor performance in most environments.

**Reply by cpwan** [reply]
I am doing RL on combinatorial problems. The prior works in this direction use REINFORCE. It would be nice if we can benchmark it.

**Reply by PaParaZz1** [reply]
Got it. I will implement naive pg in next two weeks.

**Comment by PaParaZz1** [comment]
I have implemented REINFORCE in this #544.
